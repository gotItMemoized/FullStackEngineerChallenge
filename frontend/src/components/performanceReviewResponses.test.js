import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PerformanceReviewResponses from './performanceReviewResponses';
import 'jest-dom/extend-expect';

afterEach(cleanup);

it('should render loading', () => {
  const { container } = render(<PerformanceReviewResponses />);
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('performance-reviews-loading');
  expect(innerDiv).toMatchSnapshot();
  expect(innerDiv).toHaveTextContent('Loading');
});

it('should render not render loading', () => {
  const { container } = render(<PerformanceReviewResponses review={{ feedback: [] }} />);
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('performance-reviews-loading');
  expect(innerDiv).toEqual(null);
  const innerDiv2 = document.getElementById('performance-review-response-title');
  expect(innerDiv2).toEqual(null);
});

it('should render title', () => {
  const { container } = render(
    <PerformanceReviewResponses review={{ feedback: [], user: { label: 'hi' } }} />,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('performance-reviews-loading');
  expect(innerDiv).toEqual(null);
  const innerDiv2 = document.getElementById('performance-review-response-title');
  expect(innerDiv2).toHaveTextContent('Review of hi');
});

it('should render reviews', () => {
  const { container } = render(
    <PerformanceReviewResponses
      review={{
        feedback: [
          { id: '1', message: 'test', reviewer: { name: 'j', username: 'c', id: '1' } },
          { id: '2', message: 'test2', reviewer: { name: 'j', username: 'c', id: '2' } },
        ],
        user: { label: 'hi', key: '1' },
      }}
    />,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('performance-reviews-loading');
  expect(innerDiv).toEqual(null);
  const innerDiv2 = document.getElementById('performance-review-response-title');
  expect(innerDiv2).toHaveTextContent('Review of hi');
});
