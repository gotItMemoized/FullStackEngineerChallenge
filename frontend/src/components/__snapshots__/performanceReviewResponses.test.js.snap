// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loading 1`] = `
<div>
  <div
    id="performance-reviews-loading"
  >
    Loading...
  </div>
</div>
`;

exports[`should render loading 2`] = `
<div
  id="performance-reviews-loading"
>
  Loading...
</div>
`;

exports[`should render not render loading 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;

exports[`should render reviews 1`] = `
<div>
  <div>
    <h2
      class="is-size-5 title"
      id="performance-review-response-title"
    >
      Review of 
      hi
    </h2>
    <div>
      <article
        class="message message"
      >
        <div
          class="message-body"
        >
          test
          <hr />
           - 
          j
           (
          c
          )
           
          <span
            class="tag is-info"
          >
            Self Review
          </span>
        </div>
      </article>
      <article
        class="message message"
      >
        <div
          class="message-body"
        >
          test2
          <hr />
           - 
          j
           (
          c
          )
           
        </div>
      </article>
    </div>
  </div>
</div>
`;

exports[`should render title 1`] = `
<div>
  <div>
    <h2
      class="is-size-5 title"
      id="performance-review-response-title"
    >
      Review of 
      hi
    </h2>
    <div />
  </div>
</div>
`;
